File: src/main/java/com/planit/mealplanner/model/entity/Nutrient.java
package com.planit.mealplanner.model.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.math.BigDecimal;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "nutrient")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Nutrient
{
    @Id
    @GeneratedValue
    @Column(name = "id")
    private Integer id;

    @Column(name = "name")
    private String name;

    @Column(name = "type")
    private String type;

    @Column(name = "amount")
    private Integer amount;

    @Column(name = "kcal")
    private Double kcal;

    @Column(name = "protein")
    private Double protein;

    @Column(name = "carbohydrate")
    private Double carbohydrate;

    @Column(name = "saturated_fat")
    private Double saturatedFat;

    @Column(name = "unsaturatedFat")
    private Double unsaturatedFat;

    @Column(name = "sugar")
    private Double sugar;

    @Column(name = "salt")
    private Double salt;

    @Column(name = "fiber")
    private Double fiber;

    @Column(name = "package_size")
    private Double packageSize;

    @Column(name = "package_price")
    private BigDecimal packagePrice;

    @OneToMany(mappedBy = "nutrient")
    private Set<MealPlanNutrient> mealPlanNutrient = new HashSet<>();
}

File: angular-frontend/src/app/models/nutrient.ts
import { NutrientDto } from './dto/nutrient-dto';

export class Nutrient {
  private id: number;

  private name: string;
  private type: string;
  private amount: number;
  private kcal: number;
  private protein: number;
  private carbohydrate: number;
  private saturatedFat: number;
  private unsaturatedFat: number;
  private sugar: number;
  private salt: number;
  private fiber: number;
  private packageSize: number;
  private packagePrice: number;

  constructor(dto: NutrientDto) {
    this.id = dto.id;

    this.name = dto.name;
    this.type = dto.type;
    this.amount = dto.amount;
    this.kcal = dto.kcal;
    this.protein = dto.protein;
    this.carbohydrate = dto.carbohydrate;
    this.saturatedFat = dto.saturatedFat;
    this.unsaturatedFat = dto.unsaturatedFat;
    this.sugar = dto.sugar;
    this.salt = dto.salt;
    this.fiber = dto.fiber;
    this.packageSize = dto.packageSize;
    this.packagePrice = dto.packagePrice;
  }

  getId(): number {
    return this.id;
  }

  getName(): string {
    return this.name;
  }

  getType(): string {
    return this.type;
  }

  getAmount(): number {
    return this.amount;
  }

  getKcal(): number {
    return this.kcal;
  }

  getProtein(): number {
    return this.protein;
  }

  getCarbohydrate(): number {
    return this.carbohydrate;
  }

  getSaturatedFat(): number {
    return this.saturatedFat;
  }

  getUnsaturatedFat(): number {
    return this.unsaturatedFat;
  }

  getSugar(): number {
    return this.sugar;
  }

  getSalt(): number {
    return this.salt;
  }

  getFiber(): number {
    return this.fiber;
  }

  getPackageSize(): number {
    return this.packageSize;
  }

  getPackagePrice(): number {
    return this.packagePrice;
  }
}
